src/main/java/programminglife/model/drawing/SubGraph.java

```diff
@@ -43,9 +44,10 @@
      * @param centerNode The centerNode
      * @param radius The radius
      */
-    public SubGraph(DrawableNode centerNode, int radius) {
+    public SubGraph(GenomeGraph graph, DrawableSegment centerNode, int radius) {
         // TODO
         // tactic: first go to all parents at exactly radius, then find all children of those parents
+        this.graph = graph;
```

Comment:
Either assert that the graph of the centerNode is the given graph, or just get the graph from the centerNode.